rclpy/rclpy/context.py

```diff
@@ -61,6 +61,11 @@ def init(self,
 
         if domain_id is None:
             domain_id = rclpy_implementation.rclpy_get_default_domain_id()
+        else:
+            if domain_id < 0:
+                raise RuntimeError(
+                    'Unexpected error: domain id ({}) should never be lower than zero.'
```

Comment:
```suggestion
                    'domain id ({}) should never be lower than zero.'
```

I think that this is an `expected error`.

Reply:
I refer to template of below existing codes.  

https://github.com/ros2/rclpy/blob/9b26f5647d2ab7206e66c85de2dc9776b1f608b7/rclpy/rclpy/context.py#L121-L123

Reply:
this case you pointed is actually `unexpected`, it expects the counter never goes less than zero. but with domain id, user application actually can specify the negative domain id.

Reply:
OK. Update it.